src/scenarios/shared/versionmanager.py

```diff
@@ -42,9 +42,27 @@ def get_version_from_dll_powershell_ios(dll_path: str):
 def get_runtime_commit_from_manifest(dotnet_commit_hash: str) -> str:
     """
     Fetch the actual dotnet/runtime commit SHA from the dotnet/dotnet source-manifest.json
-    :param dotnet_commit_hash: The commit hash from dotnet/dotnet repo
+    :param dotnet_commit_hash: The commit hash from dotnet/dotnet repo (or potentially already from dotnet/runtime)
     :return: The actual dotnet/runtime commit SHA, or the original hash if fetch fails
     """
+    # First check if this commit hash already exists in dotnet/runtime (backwards compatibility)
```

Comment:
Since going forwards, all SDKs will have dotnet/dotnet commit hashes, I would rather first fetch the source-manifest.json from the dotnet/dotnet repository, and then if that fails, fall back to checking if it is a valid dotnet/runtime hash.